[ISSUE] Issue with `databricks_workspace_binding` resource for storage_credential securable

### Configuration
<!-- Please provide a minimal reproducible configuration for the issue -->
```hcl
resource "databricks_workspace_binding" "readonly_bindings" {
  for_each       = toset(var.workspaceIds_with_readonly_bindings)
  securable_name = databricks_storage_credential.this.name
  workspace_id   = each.value
  securable_type = "storage_credential"
  binding_type   = "BINDING_TYPE_READ_ONLY"
}
```

### Expected Behavior
Configuration for  binding_type   = "BINDING_TYPE_READ_ONLY" works as documented for the latest version of the provider: [Link](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/workspace_binding#argument-reference)

Deployment of the storage_credential binding succeeds and enforces the read-only access for the storage credential.

### Actual Behavior
deployment fails with the following error:
```
│ Error: cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE
│
│   with module.dwh-storage-credentials.databricks_workspace_binding.readonly_bindings["<WORKSPACE_ID>"],
│   on ..\modules\storage_credentials\main.tf line 27, in resource "databricks_workspace_binding" "readonly_bindings":
│   27: resource "databricks_workspace_binding" "readonly_bindings" {
│
```

### Steps to Reproduce
terraform init
terraform plan 
terraform apply

### Terraform and provider versions
Terraform v1.13.0
on windows_amd64
+ provider registry.terraform.io/databricks/databricks v1.87.1

### Is it a regression?
Don't think so.

### Debug Output
Partially anonymized
```
2025-08-27T11:37:05.652+0200 [DEBUG] provider.terraform-provider-databricks_v1.87.1.exe: PATCH /api/2.1/unity-catalog/bindings/storage_credential/tmp-mv-cm-dwh-dev-acc
> {
>   "add": [
>     {
>       "binding_type": "BINDING_TYPE_READ_ONLY",
>       "workspace_id": ******************
>     }
>   ]
> }
< HTTP/2.0 400 Bad Request
< {
<   "details": [
<     {
<       "@type": "type.googleapis.com/google.rpc.RequestInfo",
<       "request_id": "5d422530-99c1-49bf-bdcd-b39e899ddc86",
<       "serving_data": ""
<     }
<   ],
<   "error_code": "INVALID_PARAMETER_VALUE",
<   "message": "Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE"
< }: tf_rpc=ConfigureProvider @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:38 tf_mux_provider=tf5to6server.v5tov6Server tf_req_id=d70a0238-66df-d633-e0a5-429ab22fd4cf @module=databricks tf_provider_addr=registry.terraform.io/databricks/databricks timestamp="2025-08-27T11:37:05.652+0200"
2025-08-27T11:37:05.652+0200 [DEBUG] provider.terraform-provider-databricks_v1.87.1.exe: non-retriable error: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE: tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=d70a0238-66df-d633-e0a5-429ab22fd4cf tf_rpc=ConfigureProvider @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:38 tf_mux_provider=tf5to6server.v5tov6Server @module=databricks timestamp="2025-08-27T11:37:05.652+0200"
2025-08-27T11:37:05.653+0200 [ERROR] provider.terraform-provider-databricks_v1.87.1.exe: Response contains error diagnostic: tf_rpc=ApplyResourceChange @caller=/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_severity=ERROR tf_proto_version=6.9 tf_resource_type=databricks_workspace_binding @module=sdk.proto diagnostic_summary="cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE" tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=ff0dbb4f-558c-3f8a-2a78-de2d953b9db6 timestamp="2025-08-27T11:37:05.652+0200"
2025-08-27T11:37:05.657+0200 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-08-27T11:37:05.657+0200 [ERROR] vertex "module.dwh-storage-credentials.databricks_workspace_binding.readonly_bindings[\"***********************\"]" error: cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE
2025-08-27T11:37:06.104+0200 [DEBUG] provider.terraform-provider-databricks_v1.87.1.exe: PATCH /api/2.1/unity-catalog/bindings/storage_credential/tmp-mv-cm-dwh-dev-acc
> {
>   "add": [
>     {
>       "binding_type": "BINDING_TYPE_READ_ONLY",
>       "workspace_id": *****************
>     }
>   ]
> }
< HTTP/2.0 400 Bad Request
< {
<   "details": [
<     {
<       "@type": "type.googleapis.com/google.rpc.RequestInfo",
<       "request_id": "c76b69be-1d68-4f2f-84d7-b9da42ea7f72",
<       "serving_data": ""
<     }
<   ],
<   "error_code": "INVALID_PARAMETER_VALUE",
<   "message": "Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE"
< }: tf_mux_provider=tf5to6server.v5tov6Server tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=d70a0238-66df-d633-e0a5-429ab22fd4cf tf_rpc=ConfigureProvider @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:38 @module=databricks timestamp="2025-08-27T11:37:06.104+0200"
2025-08-27T11:37:06.105+0200 [DEBUG] provider.terraform-provider-databricks_v1.87.1.exe: non-retriable error: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE: @module=databricks tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=d70a0238-66df-d633-e0a5-429ab22fd4cf @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:38 tf_mux_provider=tf5to6server.v5tov6Server tf_rpc=ConfigureProvider timestamp="2025-08-27T11:37:06.104+0200"
2025-08-27T11:37:06.105+0200 [ERROR] provider.terraform-provider-databricks_v1.87.1.exe: Response contains error diagnostic: @module=sdk.proto diagnostic_detail="" tf_proto_version=6.9 tf_req_id=d828b413-2a36-4157-52f0-213cb4cd55c7 tf_resource_type=databricks_workspace_binding tf_rpc=ApplyResourceChange @caller=/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_severity=ERROR diagnostic_summary="cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE" tf_provider_addr=registry.terraform.io/databricks/databricks timestamp="2025-08-27T11:37:06.104+0200"
2025-08-27T11:37:06.107+0200 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-08-27T11:37:06.107+0200 [ERROR] vertex "module.dwh-storage-credentials.databricks_workspace_binding.readonly_bindings[\"*****************\"]" error: cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE
╷
│ Error: cannot create workspace binding: Supported binding types for STORAGE_CREDENTIAL are BINDING_TYPE_READ_WRITE
│
│   with module.dwh-storage-credentials.databricks_workspace_binding.readonly_bindings["*****************"],
│   on ..\modules\storage_credentials\main.tf line 27, in resource "databricks_workspace_binding" "readonly_bindings":
│   27: resource "databricks_workspace_binding" "readonly_bindings" {
│
╵
```
### Important Factoids

### Would you like to implement a fix?
No idea how to do this.

--- a/NEXT_CHANGELOG.md
+++ b/NEXT_CHANGELOG.md
@@ -14,6 +14,8 @@
 ### Documentation
 * Added `disabled` field to `task` block in `databricks_job` resource, allowing individual tasks to be disabled ([#5767](https://github.com/databricks/terraform-provider-databricks/pull/5767)).
 
+* Document that read-only workspace bindings aren't applicable for non-catalog objects ([#5611](https://github.com/databricks/terraform-provider-databricks/pull/5611))
+
 ### Exporter
 
 ### Internal Changes

--- a/docs/resources/workspace_binding.md
+++ b/docs/resources/workspace_binding.md
@@ -36,7 +36,7 @@ The following arguments are required:
 * `workspace_id` - ID of the workspace. Change forces creation of a new resource.
 * `securable_name` - Name of securable. Change forces creation of a new resource.
 * `securable_type` - Type of securable. Can be `catalog`, `external_location`, `storage_credential` or `credential`. Default to `catalog`. Change forces creation of a new resource.
-* `binding_type` - (Optional) Binding mode. Default to `BINDING_TYPE_READ_WRITE`. Possible values are `BINDING_TYPE_READ_ONLY`, `BINDING_TYPE_READ_WRITE`.
+* `binding_type` - (Optional) Binding mode. Default to `BINDING_TYPE_READ_WRITE`. Possible values are `BINDING_TYPE_READ_ONLY`, `BINDING_TYPE_READ_WRITE`. Note: `BINDING_TYPE_READ_ONLY` is only supported when `securable_type` is `catalog`. For `storage_credential`, `external_location`, and `credential`, only `BINDING_TYPE_READ_WRITE` is supported.
 * `provider_config` - (Optional) Configure the provider for management through account provider. This block consists of the following fields:
   * `workspace_id` - (Required) Workspace ID which the resource belongs to. This workspace must be part of the account which the provider is configured with.
 

